.github/workflows/book.yml

```diff
@@ -20,9 +33,26 @@ jobs:
         working-directory: book
       - run: cp -r book/book ../book
       - run: git checkout -- . && git clean -fd
-      - run: git fetch && git checkout gh-pages
       - run: test ! -d $(echo ${{github.ref}} | cut -d/ -f3) || rm -r $(echo ${{github.ref}} | cut -d/ -f3)
       - run: cp -r ../book $(echo ${{github.ref}} | cut -d/ -f3)
-      - run: git config --local user.name "github-actions[bot]" && git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
-      - run: git add $(echo ${{github.ref}} | cut -d/ -f3) && git commit --allow-empty -m "Docs build for SOF3/await-generator@${{github.sha}}"
-      - run: git push
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./
+          destination: ./_site
```

Comment:
Does this mean we cannot persist pages on other branches/tags?

Reply:
Let me add a download artifact action.

Reply:
@Endermanbugzjfc artifact of what? previous action run?

Reply:
> What is the point of spending so much effort to manually manage the previous artifact versions when the current workflow already works, using Git VCS to help us manage multiple versions?

For faster clone speed.

Reply:
Well, there is still a problem tho. The current [README](https://sof3.github.io/await-generator/) is not up to date.